gradio/components.py

```diff
@@ -2550,10 +2565,28 @@ def __validate_headers(headers: List[str] | None, col_count: int):
                 )
             )
 
+    @staticmethod
+    def __process_markdown(data: List[List[Any]], datatype: str | List[str]):
+        if (
```

Comment:
@pngwn I think I figured out the bug.

In the demo `datatype` is `['markdown', 'markdown', 'html']`. That's always not equal to the string "markdown" so the `or` evaluates to True.

A more verbose but easier to understand implementation would be

```python
        if (
            isinstance(datatype, list) and "markdown" not in datatype
        ):
            return data
        if isinstance(datatype, str) and datatype != "markdown":
            return data
```

Also `isinstance` is better that `type(....) is ...` in my opinion

Reply:
It can render the markdown now

<img width="1344" alt="image" src="https://user-images.githubusercontent.com/41651716/176918894-ce424308-387f-4fca-86a3-38679dfe8d30.png">

Reply:
Happy to change, google guides me.